pkg/frontend/validate.go

```diff
@@ -44,3 +44,22 @@ func (f *frontend) validateSubscriptionState(ctx context.Context, key string, al
 
 	return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionState, "", "Request is not allowed in subscription in state '%s'.", doc.Subscription.State)
 }
+
+// validateOpenShiftUniqueKey returns which unique key if causing a 412 error
+func (f *frontend) validateOpenShiftUniqueKey(ctx context.Context, doc *api.OpenShiftClusterDocument) error {
+	docs, err := f.db.OpenShiftClusters.ValidateUniqueKey(ctx, doc.PartitionKey, "clusterResourceGroupIdKey", doc.ClusterResourceGroupIDKey)
+	if err != nil {
+		return err
+	}
+	if docs.Count != 0 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeDuplicateResourceGroup, "", "The provided resource group '%s' already contains a cluster.", doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID)
+	}
+	docs, err = f.db.OpenShiftClusters.ValidateUniqueKey(ctx, doc.PartitionKey, "clientIdKey", doc.ClientIDKey)
+	if err != nil {
+		return err
+	}
+	if docs.Count != 0 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeDuplicateClientID, "", "Each ARO cluster must use an unique SPN and cannot be shared with other clusters. Please use a new service principal.")
```

Comment:
`return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeDuplicateClientID, "", "The provided client ID '%s' is already in use by a cluster.", doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID)` ?